flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adapter/DefaultExecutionTopology.java

```diff
@@ -243,13 +275,36 @@ private static DefaultExecutionVertex generateSchedulingExecutionVertex(
     }
 
     private static IndexedPipelinedRegions computePipelinedRegions(
-            Iterable<DefaultExecutionVertex> topologicallySortedVertexes,
+            Iterable<DefaultLogicalPipelinedRegion> logicalPipelinedRegions,
+            Function<DefaultLogicalPipelinedRegion, List<DefaultExecutionVertex>>
+                    sortedExecutionVerticesInPipelinedRegion,
+            Function<ExecutionVertexID, DefaultExecutionVertex> executionVertexRetriever,
             Function<IntermediateResultPartitionID, DefaultResultPartition>
                     resultPartitionRetriever) {
+
         long buildRegionsStartTime = System.nanoTime();
 
         Set<Set<SchedulingExecutionVertex>> rawPipelinedRegions =
-                PipelinedRegionComputeUtil.computePipelinedRegions(topologicallySortedVertexes);
+                Collections.newSetFromMap(new IdentityHashMap<>());
+
+        for (DefaultLogicalPipelinedRegion logicalPipelinedRegion : logicalPipelinedRegions) {
+
+            List<DefaultExecutionVertex> schedulingExecutionVertices =
+                    sortedExecutionVerticesInPipelinedRegion.apply(logicalPipelinedRegion);
+
+            if (containsIntraRegionAllToAllEdge(logicalPipelinedRegion)) {
+                // If the logical pipelined region contains any intra-region all-to-all edge,
+                // convert the entire logical pipelined region to the sole scheduling pipelined
+                // region
+                rawPipelinedRegions.add(new HashSet<>(schedulingExecutionVertices));
+            } else {
+                rawPipelinedRegions.addAll(
```

Comment:
Let's add some explanation for why we are computing scheduling pipelined regions out from vertices of a logical pipelined region.